### PR TITLE
samba: add python dependencies to buildInputs

### PR DESCRIPTION
**Pull request by amaxine**
###### Motivation for this change
```
package = pkgs.samba4.override {
  enableLDAP = true;
  enablePrinting = true;
  enableMDNS = true;
  enableDomainController = true;
  enableRegedit = true;
};
```

I couldn't build Samba after the most recent bump (#124693), this fixed things. It seems both are only needed for AD, I haven't tested building it in other configurations than above.

I should probably trim my configuration - I don't think I actually need ldap support at all here :sweat_smile: - but I thought this would be worth PRing regardless. There might be more breakage elsewhere, and I imagine this needs backporting same as the other PR.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125105` at c8eeabbf run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/6d35141a50219029786ddbb60fda91ae">sambaFull</a> (samba4Full)</li>
  </ul>
</details>
<details>
  <summary>5 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/samba/4.x.nix:58:5:
```
   |
58 |     ./patch-source3__libads__kerberos_keytab.c.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/servers/samba/4.x.nix:157:5:
```
    |
157 |     license = licenses.gpl3;
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/samba/4.x.nix:59:5:
```
   |
59 |     ./4.x-no-persistent-install-dynconfig.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/samba/4.x.nix:57:5:
```
   |
57 |     ./4.x-no-persistent-install.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/samba/4.x.nix:60:5:
```
   |
60 |     ./4.x-fix-makeflags-parsing.patch
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by mweinelt** [comment]
Hi, sorry for being so pushy, this is also broken on release-21.05 and I'm in a hurry to get that fixed.

The package still failed to build, so here is what I changed:

- Move `python3Packages.dnspython` over into the domain controller section, it is used in a dnsupdate script, not part of LDAP
- Fixed LDAP by including `openldap.dev` output, instead ouf the default `openldap.out`
- Finally remove `enableGlusterFS` from `samba4Full` to unbreak the package

**Comment by maxeaubrey** [comment]
No worries. I placed `dnspython` under LDAP based on docs more than anything, but I could have just misread those. Thanks for fixing the rest up, including the full package!

File: pkgs/servers/samba/4.x.nix [hunk 1]
```diff
@@ -91,10 +91,10 @@ stdenv.mkDerivation rec {
     libtasn1
     tdb
   ] ++ optionals stdenv.isLinux [ liburing systemd ]
-    ++ optional enableLDAP openldap
+    ++ optional enableLDAP [ openldap.dev python3Packages.markdown ]
```
Comment (SuperSandro2000) [inline pkgs/servers/samba/4.x.nix:94]:
```suggestion
    ++ optionals enableLDAP [ openldap.dev python3Packages.markdown ]
```

Comment (mweinelt) [reply pkgs/servers/samba/4.x.nix]: Thanks, fixed.

**Pull request merged**